File: bank-service/src/main/java/rs/edu/raf/mapper/CountryMapper.java
package rs.edu.raf.mapper;

import org.springframework.stereotype.Component;
import rs.edu.raf.dto.CountryCreateDTO;
import rs.edu.raf.dto.CountryDTO;
import rs.edu.raf.model.Country;

@Component
public class CountryMapper {

    public Country countryCreateDTOtoCountry(CountryCreateDTO countryCreateDTO) {
        Country country = new Country();

        country.setName(countryCreateDTO.name());

        return country;
    }

    public CountryDTO countryToCountryDTO(Country country) {
        return new CountryDTO(country.getId(), country.getName());
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/mapper/CountryMapper.kt
package org.example.bankservicekotlin.mapper

import org.example.bankservicekotlin.dto.CountryDTO
import org.example.bankservicekotlin.dto.CountryCreateDTO
import org.example.bankservicekotlin.model.Country
import org.springframework.stereotype.Component

@Component
class CountryMapper {
    fun countryCreateDTOtoCountry(countryCreateDTO: CountryCreateDTO): Country {
        return Country(null, countryCreateDTO.name)
    }

    fun countryToCountryDTO(country: Country): CountryDTO {
        return CountryDTO(country.id, country.name)
    }
}